python/pyspark/sql/tests/test_pandas_udf_grouped_agg.py

```diff
@@ -491,6 +491,28 @@ def max_udf(v):
             agg2 = self.spark.sql("select max_udf(id) from table")
             assert_frame_equal(agg1.toPandas(), agg2.toPandas())
 
+    def test_no_predicate_pushdown_through(self):
+        from pyspark.sql.functions import monotonically_increasing_id, explode_outer
+        import numpy as np
+
+        @pandas_udf('float', PandasUDFType.GROUPED_AGG)
+        def mean(x):
+            return np.mean(x)
+
+        df = self.spark.createDataFrame([
+            Row(foo=[Row(bar=42), Row(bar=43), Row(bar=44)]),
+        ])
+
+        df_with_id = df.withColumn('id', monotonically_increasing_id())
+        exploded = df_with_id.select('id', explode_outer('foo').alias('foos'))
```

Comment:
@viirya, is it necessary to use `monotonically_increasing_id` and `explode_outer`?  I know it's extracted from the reported case but I think we should just make it minimised for the fix.

Reply:
Ah, this comes from reported test code. Let me simplify it a bit. :)